### ISSUE TITLE
[20190122-17:35:40] [DEBUG] errno: 2, description: No such file or directory

### ISSUE THREAD

**Issue by tps800** [neutrinolabs/xrdp#1281]
**OS:** CenoOS 6.9 on i386
**xrdp:** xrdp-0.6.1-4.el6.i686

Trying to start sessman-x11rdp session fails without any hint on the terminal side why it fails. Login is ok, short after `mstcs` closes. Within `/var/log/xrdp-sesman.log` I find:
```
[20190122-17:35:40]` [INFO ] scp thread on sck 7 started successfully
[20190122-17:35:40] [INFO ] ++ created session (access granted): username imisuser, ip 10.160.2.12:52867 - socket: 7
[20190122-17:35:40] [INFO ] starting X11rdp session...
[20190122-17:35:40] [CORE ] error starting X server - user testuser - pid 18314
[20190122-17:35:40] [DEBUG] errno: 2, description: No such file or directory
[20190122-17:35:40] [DEBUG] execve parameter list: 10
[20190122-17:35:40] [DEBUG]         argv[0] = X11rdp
[20190122-17:35:40] [DEBUG]         argv[1] = :11
[20190122-17:35:40] [DEBUG]         argv[2] = -geometry
[20190122-17:35:40] [DEBUG]         argv[3] = 1280x1024
[20190122-17:35:40] [DEBUG]         argv[4] = -depth
[20190122-17:35:41] [DEBUG]         argv[5] = 24
[20190122-17:35:41] [DEBUG]         argv[6] = -bs
[20190122-17:35:41] [DEBUG]         argv[7] = -nolisten
[20190122-17:35:41] [DEBUG]         argv[8] = tcp
[20190122-17:35:41] [DEBUG]         argv[9] = (null)
[20190122-17:35:50] [ERROR] X server for display 11 startup timeout
[20190122-17:35:50] [INFO ] starting xrdp-sessvc - xpid=18314 - wmpid=18313
[20190122-17:35:50] [ERROR] X server for display 11 startup timeout
[20190122-17:35:50] [ERROR] another Xserver is already active on display 11
[20190122-17:35:50] [DEBUG] aborting connection...
[20190122-17:35:50] [INFO ] ++ terminated session:  username testuser, display :11.0, session_pid 18312, ip 10.160.2.12:52867 - socket: 7
```

It cant find some unmentioned file. My question: which file is not found?

**Configuration `/etc/xrdp/xrdp.ini`:**
```
[globals]
bitmap_cache=yes
bitmap_compression=yes
port=3389
crypt_level=high
channel_code=1
max_bpp=24
#black=000000
#grey=d6d3ce
#dark_grey=808080
#blue=08246b
#dark_blue=08246b
#white=ffffff
#red=ff0000
#green=00ff00
#background=626c72

[xrdp1]
name=sesman-X11rdp
lib=libxup.so
username=ask
password=ask
ip=127.0.0.1
port=-1
xserverbpp=24

[xrdp2]
name=sesman-Xvnc
lib=libvnc.so
username=ask
password=ask
ip=127.0.0.1
port=-1
```

**`/etc/xrdp/sesman.ini`:**
```
[Globals]
ListenAddress=127.0.0.1
ListenPort=3350
EnableUserWindowManager=1
UserWindowManager=startwm.sh
DefaultWindowManager=startwm.sh

[Security]
AllowRootLogin=1
MaxLoginRetry=4
TerminalServerUsers=tsusers
TerminalServerAdmins=tsadmins

[Sessions]
X11DisplayOffset=10
MaxSessions=10
KillDisconnected=0
IdleTimeLimit=0
DisconnectedTimeLimit=0

[Logging]
LogFile=/var/log/xrdp-sesman.log
LogLevel=DEBUG
EnableSyslog=0
SyslogLevel=DEBUG

[X11rdp]
param1=-bs
param2=-nolisten
param3=tcp

[Xvnc]
param1=-bs
param2=-nolisten
param3=tcp
param4=-localhost
param5=-dpi
param6=96

[Xorg]
param1=/usr/bin/Xorg
```

**The package installed:**
```
/etc/logrotate.d/xrdp
/etc/pam.d/xrdp-sesman
/etc/rc.d/init.d/xrdp
/etc/sysconfig/xrdp
/etc/xrdp
/etc/xrdp/km-0407.ini
/etc/xrdp/km-0409.ini
/etc/xrdp/km-040c.ini
/etc/xrdp/km-0410.ini
/etc/xrdp/km-0419.ini
/etc/xrdp/km-041d.ini
/etc/xrdp/rsakeys.ini
/etc/xrdp/sesman.ini
/etc/xrdp/startwm.sh
/etc/xrdp/xrdp.ini
/etc/xrdp/xrdp.sh
/usr/bin/xrdp-dis
/usr/bin/xrdp-genkeymap
/usr/bin/xrdp-keygen
/usr/bin/xrdp-sesadmin
/usr/bin/xrdp-sesrun
/usr/bin/xrdp-sestest
/usr/lib/xrdp
/usr/lib/xrdp/libcommon.so
/usr/lib/xrdp/libcommon.so.0
/usr/lib/xrdp/libcommon.so.0.0.0
/usr/lib/xrdp/libmc.so
/usr/lib/xrdp/libmc.so.0
/usr/lib/xrdp/libmc.so.0.0.0
/usr/lib/xrdp/librdp.so
/usr/lib/xrdp/librdp.so.0
/usr/lib/xrdp/librdp.so.0.0.0
/usr/lib/xrdp/libscp.so
/usr/lib/xrdp/libscp.so.0
/usr/lib/xrdp/libscp.so.0.0.0
/usr/lib/xrdp/libvnc.so
/usr/lib/xrdp/libvnc.so.0
/usr/lib/xrdp/libvnc.so.0.0.0
/usr/lib/xrdp/libxrdp.so
/usr/lib/xrdp/libxrdp.so.0
/usr/lib/xrdp/libxrdp.so.0.0.0
/usr/lib/xrdp/libxup.so
/usr/lib/xrdp/libxup.so.0
/usr/lib/xrdp/libxup.so.0.0.0
/usr/sbin/xrdp
/usr/sbin/xrdp-chansrv
/usr/sbin/xrdp-sesman
/usr/sbin/xrdp-sessvc
/usr/share/doc/xrdp-0.6.1
/usr/share/doc/xrdp-0.6.1/COPYING
/usr/share/doc/xrdp-0.6.1/design.txt
/usr/share/doc/xrdp-0.6.1/faq-compile.txt
/usr/share/doc/xrdp-0.6.1/faq-general.txt
/usr/share/doc/xrdp-0.6.1/file-loc.txt
/usr/share/doc/xrdp-0.6.1/install.txt
/usr/share/doc/xrdp-0.6.1/prog_std.txt
/usr/share/doc/xrdp-0.6.1/readme.txt
/usr/share/man/man5/sesman.ini.5.gz
/usr/share/man/man5/xrdp.ini.5.gz
/usr/share/man/man8/xrdp-sesman.8.gz
/usr/share/man/man8/xrdp-sesrun.8.gz
/usr/share/man/man8/xrdp.8.gz
/usr/share/xrdp
/usr/share/xrdp/ad24b.bmp
/usr/share/xrdp/ad256.bmp
/usr/share/xrdp/cursor0.cur
/usr/share/xrdp/cursor1.cur
/usr/share/xrdp/sans-10.fv1
/usr/share/xrdp/xrdp24b.bmp
/usr/share/xrdp/xrdp256.bmp
/var/log/xrdp-sesman.log
```

Selecting `sesman-Xvnc` at login works. But it would be nice this would be true for `sesman-X11rdp` too.
Something must be missing from this installation (all 109 Clients have the very same problem). But since xrdp does not mention which file it is missing and where it is searching for it, I'll consider this a bug ....

At the time being, no newer xrdp-package than 0.6.1-4 is available for CentOS 6.9.

**Comment by tps800** [comment]
A package xorgxrdp is not available for CentOS 6.9, but for CentOS 7.x -- maybe this package is missing to make sesman-x11rdp work?

**Comment by metalefty** [comment]
xrdp 0.6.x is now out of support.
